x-pack/plugin/core/src/main/java/org/elasticsearch/xpack/core/security/authz/RoleDescriptor.java

```diff
@@ -485,8 +556,53 @@ private static RoleDescriptor.IndicesPrivileges[] parseIndices(String roleName,
         return privileges.toArray(new IndicesPrivileges[privileges.size()]);
     }
 
-    private static RoleDescriptor.IndicesPrivileges parseIndex(String roleName, XContentParser parser, boolean allow2xFormat)
+    private static IndicesPrivileges parseIndex(final String roleName, final XContentParser parser, final boolean allow2xFormat)
+        throws IOException {
+        final Tuple<IndicesPrivileges, String[]> parsed = parseIndexWithOptionalRemoteClusters(roleName, parser, allow2xFormat, false);
+        assert parsed.v2() == null : "indices privileges cannot have remote clusters";
+        return parsed.v1();
+    }
+
+    private static RoleDescriptor.RemoteIndicesPrivileges[] parseRemoteIndices(
+        final String roleName,
+        final XContentParser parser,
+        final boolean allow2xFormat
+    ) throws IOException {
+        if (parser.currentToken() != XContentParser.Token.START_ARRAY) {
+            throw new ElasticsearchParseException(
+                "failed to parse remote indices privileges for role [{}]. expected field [{}] value "
+                    + "to be an array, but found [{}] instead",
+                roleName,
+                parser.currentName(),
+                parser.currentToken()
+            );
+        }
+        final List<RoleDescriptor.RemoteIndicesPrivileges> privileges = new ArrayList<>();
+        while (parser.nextToken() != XContentParser.Token.END_ARRAY) {
+            privileges.add(parseRemoteIndex(roleName, parser, allow2xFormat));
+        }
+        return privileges.toArray(new RemoteIndicesPrivileges[0]);
+    }
+
+    private static RemoteIndicesPrivileges parseRemoteIndex(String roleName, XContentParser parser, boolean allow2xFormat)
```

Comment:
There is no need to accept 2.x format anymore for the new remote indices privileges.